doc/neps/nep-0031-uarray.rst

```diff
@@ -69,12 +69,28 @@ _will be discussed on the mailing list._
 
 The way we propose the overrides will be used by end users is::
 
-    from numpy import unumpy
-    TODO
+    from numpy import unumpy as np
+    with np.set_backend(backend):
+        x = np.asarray(my_array, dtype=dtype)
 
-And a library that implements a NumPy-like API will use it like::
+And a library that implements a NumPy-like API will use it in the following manner (as an example)::
 
-    TODO: example corresponding to NEP 30 `duckarray`
+    _ua_implementations = {}
+
+    def __ua_function__(func, args, kwargs):
+        return _ua_implementations[func](*args, **kwargs)
+
+    def implements(ua_func):
+        def inner(func):
+            _ua_implementations[ua_func] = func
+            return func
+
+        return inner
+
+    @implements(np.asarray)
+    def asarray(a, dtype=None, order=None):
+        # Code here
+        # Must return NotImplemented for unsupported array types
```

Comment:
Just to note that this is not what `dask.array.asarray` does, it still contains a fallback to `np.asarray`. _Should_ dask change that, or _must_ it change that?

Reply:
It will work if it doesn't change it, but that isn't what it should do otherwise other backends won't even be tried. So for proper functioning I'd say it's a must...

Reply:
To be precise: either `__ua_convert__` or `__ua_function__` *must* return `NotImplemented`. Returning it from `asarray` would be one way to satisfy that requirement.

Reply:
For something like `asarray` whose defined function is to coerce I think it's fine if it coerces, but failure should be `NotImplemented`. Consider jagged arrays for example, which `xnd` can handle fine.